pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultipleOrEqualitiesToInClauseFilterQueryTreeOptimizer.java

```diff
@@ -42,7 +43,7 @@ public FilterQueryTree optimize(FilterQueryOptimizerRequest request) {
 
   private FilterQueryTree optimize(FilterQueryTree filterQueryTree, FilterQueryTree parent) {
     if (filterQueryTree.getOperator() == FilterOperator.OR) {
-      Map<String, Set<String>> columnToValues = new HashMap<>();
+      Map<String, Set<String>> columnToValues = new TreeMap<>();
```

Comment:
Would be nice to add a comment on why TreeMap (ordering) is needed.

Reply:
It's not needed anymore after rewrote the comparison method.